lib/tacos/dependent_slices.py

```diff
@@ -83,6 +84,41 @@ def ls_files(self, dir: Dir) -> Iterable[File]:
                 yield file
 
 
+@dataclass(frozen=True)
+class PathFilter:
+    """A frozen view of a collection of files."""
+
+    allowed: frozenset[re.Pattern[str]]
+
+    def match(self, path: str) -> bool:
+        for pattern in self.allowed:
+            # intentionally uses match to check from the start of the string
+            if pattern.match(path) is not None:
+                return True
+        return False
+
+    @classmethod
+    def from_config(cls, path: OSPath) -> PathFilter:
+        """Get a list of allowed RE patterns from a file.
+
+        # comments are removed and blank lines are ignored.
+        An empty or missing file is treated as all allowed."""
+        lines: list[re.Pattern[str]] = []
+        # remove from the first unescaped # to the end
+        comment_pattern = re.compile(r"(?<!\\)#.*")
+        try:
+            with path.open() as config:
+                for line in config:
+                    line = comment_pattern.sub("", line).strip()
+                    if line:
+                        lines.append(re.compile(line))
```

Comment:
this is a pattern i've used for many years:

```
for line in config:
  line = line.strip()
  if not line or line.startswith('#'): continue
```

this won't give end-of-line comments but it's good enough, and regex cause bugs too often

Reply:
OTOH what you have seems right